streaming/src/main/scala/io/gearpump/streaming/task/TaskActor.scala

```diff
@@ -344,6 +354,7 @@ object TaskActor {
     def generateAckResponse(ackRequest: AckRequest, sender: ActorRef): Ack = {
       val sessionId = ackRequest.sessionId
       if (receivedMsgCount.containsKey(sessionId)) {
+        receivedMsgCount.put(sessionId, (receivedMsgCount.get(sessionId) + 1).toShort)
```

Comment:
Yes, we should do the same at the sender side.

The purpose is to reduce the AckRequest count.